### DISCUSSION TITLE
Disable annoying tip/hint box

### DISCUSSION THREAD

**Discussion by kaveh-w0** [geany/geany#3216] [Q&A]
When writing a program in C, the moment I type the opening parenthesis of printf function I get a very annoying little tip/hint box just below the function about its arguments.

printf(
       `int printf (const char * __ format, ...)`

And it won't go away unless I push up/down button on my keyboard or click somewhere on the screen or type the closing parenthesis. I need to disable it.
Geany 1.36 on ubuntu 20.04

**Comment by elextr** [comment]
Its called a "calltip" and there does not seem to be any way of disabling it for C like languages.  Pull requests adding a setting welcome.

**Comment by sk2449** [comment]
Might come a bit late, but you can start Geany with argument "--no-ctags" to disable autocompletion and tooltips. Other than the name of the argument suggests, this will disable loading of all [default global tag files](https://www.geany.org/manual/current/index.html#default-global-tags-files), not just for C-based languages.
Ran into the same annoyance with Python, when Geany even showed me the call syntax of [xmlrpc.client.loads](https://docs.python.org/3/library/xmlrpc.client.html#xmlrpc.client.loads) after typing "json.loads(". (<- worth a bug report?)

**Comment by elextr** [comment]
> but you can start Geany with argument "--no-ctags" to disable autocompletion and tooltips.

Well yes, but that also disables library autocompletion as well.

> Geany even showed me the call syntax of [xmlrpc.client.loads](https://docs.python.org/3/library/xmlrpc.client.html#xmlrpc.client.loads) after typing "json.loads(". (<- worth a bug report?)

I'm sure you get more than that?  Note the up arrow and down arrow buttons to cycle through the options and `loads(s, *, cls=None, object_hook=None, parse_float=None, parse_int=None, parse_constant=None, object_pairs_hook=None, **kw)` is in the list along with lots more.

The matching of functions to provide calltips is purely name sensitive (ie "loads") not context sensitive (ie "json.loads") so not a bug, its simply not implemented.  It is an "all languages are C" side effect (C does not contextualise functions).  Whilst you could submit a feature request, its likely that it won't be implemented and you will be referred to the LSP support instead for accurate language dependent results.
